tmail-backend/imap-extensions/src/main/java/com/linagora/tmail/imap/TMailFullDomainPathConverter.java

```
@@ -0,0 +1,115 @@
+import org.apache.james.mailbox.MailboxSession;
+import org.apache.james.mailbox.model.MailboxConstants;
+import org.apache.james.mailbox.model.MailboxPath;
+
+import java.util.List;
+import java.util.Optional;
+
+public class TMailFullDomainPathConverter extends TMailPathConverter implements PathConverter {
+    public static class Factory implements PathConverter.Factory {
+        public TMailFullDomainPathConverter forSession(ImapSession session) {
+            return new TMailFullDomainPathConverter(session.getMailboxSession());
+        }
+
+        public TMailFullDomainPathConverter forSession(MailboxSession session) {
+            return new TMailFullDomainPathConverter(session);
+        }
+    }
+
+    protected static final String LOCAL_DOMAIN = "local";
+    protected final Escaper domainEscaper;
+
+    private TMailFullDomainPathConverter(MailboxSession mailboxSession) {
+        super(mailboxSession);
+        this.domainEscaper = Escapers.builder()
+                .addEscape(mailboxSession.getPathDelimiter(), "__")
+                .addEscape('_', "_-")
+                .build();
+    }
+
+    public Optional<String> mailboxName(boolean relative, MailboxPath path, MailboxSession session) {
+        if (path.getNamespace().equalsIgnoreCase(TeamMailboxNameSpace.TEAM_MAILBOX_NAMESPACE())) {
+            // Because internal and external representations of a team mailbox differ, we need to convert
+            // the local path like MailboxPath(namespace="#Teammailbox", user="team-mailbox@<domain>", name="<teamXXX>/<folder>")
+            // to the external representation '#TeamMailbox/<teamXXX>@<domain>/<folder>'.
+            List<String> mailboxNameParts = Splitter.on(session.getPathDelimiter()).splitToList(path.getName());
+            String team = mailboxNameParts.getFirst();
+            String domain = domainEscaper.escape(
+                    path.getUser().getDomainPart().map(Domain::asString).orElse(LOCAL_DOMAIN)
```

I feel that using the "local" value here is rather forced (although this case may not actually occur in practice)
creating a user like "team-mailbox@local", a user that has never existed before, a non-existent user—could pose potential risks in the future. 
And the name "local" makes quite confused.
We can absent domain value when domainpart absent?

---

I agree, that's a bit awkward. What I don't know is - why *can* the domain of a user even be empty? In what case would that make sense? And what's the semantics when using the resulting mailbox path to search for mailboxes? Would it return mailboxes of *all* domains as long as the username matches? Or maybe nothing?

For TeamMailboxes, specifically, we always expect the user to be `team-mailbox@<domain>`. Maybe we just throw an exception if the domain is missing?

---

>  why can the domain of a user even be empty?

Maybe in case single domain. 
As document: 
```
A username can have only a local part when virtualHosting is disabled. E.g.'myUser'
```
https://james.apache.org/server/config-users.html

> For TeamMailboxes, specifically, we always expect the user to be team-mailbox@<domain>. Maybe we just throw an exception if the domain is missing?

It ok for only "TMailFullDomainPathConverter"

---

I've made the `TMailFullDomainPathConverter` now refuse missing domains.